gobblin-core-base/src/main/java/gobblin/crypto/EncryptionConfigParser.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.crypto;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+
+
+/**
+ * Extract encryption related information from taskState
+ */
+public class EncryptionConfigParser {
+  private static final String ENCRYPT_PREFIX = ".encrypt";
+
+  // TODO these are dupe with ConfigurationKeys - this is to eventually allow for encryption plugins down the road
+  // that just get handed a set of config and can decide to define their own config below the branch prefix if
+  // they want
+  public static final String ENCRYPTION_ALGORITHM_KEY = "algorithm";
```

Comment:
We no longer strictly put configuration keys in `ConfigurationKeys` (otherwise that class would get huge). I'd recommend removing them from there.